### PR TITLE
UPDATED: Adding capability to deconvolve using channel-by-channel templates

### PR DESCRIPTION
**Pull request by vpec0**
This is a reopening of #61. 

#61 was merged, caused issues, and reverted in develop.

This PR contains the same commits and additional commits which should resolve the issues in CI tests. FHICL configurations were updated so that they were backwards compatible.

However, FD reco workflows should switch to using table `dune_fd_deconvolution` from `Deconvolution.fcl` and ProtoDUNE HD should switch to table `protodunehd_deconvolution` or `protodunehd_mc_deconvolution`.

CI tests should be run before the pull, I haven't tested this on FD sims.

### CODE REVIEW

**Comment by absolution1** [comment]
> trigger build

Darn it.  @aolivier23 how do I trigger the CI within dune's organisation?

**Comment by jroto** [comment]
Thanks @vpec0, can you make a PR in dunesw with the changes you mention?

**Comment by vpec0** [comment]
> ❌ CI build for DUNE Failed at phase ci_tests DUNE on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/view_builds/index?offset=0&builds=dune_ci/18149&builds=)
> 
> 🚨 For more details about the failed phase, check the [ci_tests DUNE phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:DUNE/build_detail/phase_details?build_id=dune_ci/18149&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)
> 
> parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/24168&builds=)

There were some configurable name changes which I had not propagated fully. This is fixed in the latest commit (72e0166).

I have tested building the code and run successfully the reco job used in the CI tests:
lar -n 1 -c ci_test_reco_dunefd.fcl -o prodgenie_nue_dune10kt_1x2x6_reco_Current.root -s xroot://fndca1.fnal.gov:1095/pnfs/fnal.gov/usr/dune/persistent/stash/ContinuousIntegration/DUNEFD/detsim/prodgenie_nue_dune10kt_1x2x6_detsim_Reference.root

**Comment by jroto** [comment]
I thought @YoannKermaidic got rid of the cold box's deconvolution. As they don't have such a big undershoot, they don't need to deconvolve the signals. Yes, I wouldn't touch those.

If I understood correctly, this deconvolution module has been already tested with data, although not with the new simulation digitizer.
Also, the deconvolution of the full streaming channel is pending. Can you confirm these two points, @vpec0?

Then we need to wait for @mjdelgadog to update the SPE templates of all channels (she told me we would have it by next Monday).

Should we wait for Maritza before merging?

**Comment by vpec0** [comment]
I can confirm, that the code was not tested with PDHD sims, only with data.

Treatment of full streaming channels is not implemented and those channels should be ignored.

This PR does not need to wait for the templates to be ready. This is functionality implementation and configuration for reading out the templates from stashcache can be added later.

File: duneopdet/OpticalDetector/Deconvolution/Deconvolution.fcl [hunk 1]
```diff
@@ -8,48 +10,118 @@ WfmWienerfilter: {
...
-#Postfilter Cutoff 
+
+#Postfilter Cutoff
 dune_deconvolution.WfmPostfilter.Cutoff: 1.8     # Use this value only for postfilter.
 
+
+# For backwards compatibility. This table is currently used in many workflows for different detectors, data and MC.
+dune_deconvolution: @local::generic_dune_deconvolution
+dune_deconvolution.SPETemplateFiles: ["SPE_DAPHNE2_FBK_2022.dat"]
+# Set SPE template channel map: channel = -1 for all channels
```
Comment (lpaulucc) [inline duneopdet/OpticalDetector/Deconvolution/Deconvolution.fcl:65]: Great!

**Review by lpaulucc — APPROVED** [review]
This looks good, thanks, Viktor.

**Comment by jroto** [comment]
Can we merge this already? should we run again the CI tests? @absolution1 @aolivier23

**Comment by lpaulucc** [comment]
trigger build

**Review by absolution1 — APPROVED** [review]
Looks good, thank you

**Comment by vpec0** [comment]
I can't interpret the results from the CI tests. From a quick look, I don't see how the warnings and failures are related to the PR.

**Comment by absolution1** [comment]
@aolivier23 could say more but I think the last reference update failed and needed to be investigated.

My feeling is that that we can proceed with merging the PR.  The CI successfully ran ,so shows that the workflow mechanically runs.  This is already a valuable result.

**Review by mantheys — APPROVED** [review]

**Comment by mantheys** [comment]
Looks good to me! Does the PDS workflow validation run automatically after the PR? It would be important to make sure that the standard configuration is still backward compatible with the latest results.

**Comment by jroto** [comment]
@aolivier23, we are missing your review!

**Pull request merged**